packages/hoppscotch-sh-admin/src/composables/useConfigHandler.ts

```diff
@@ -337,28 +153,44 @@ export function useConfigHandler(updatedConfigs?: Config) {
   };
 
   // Transforming the working configs back into the format required by the mutations
-  const updatedAllowedAuthProviders = computed(() => {
-    return [
+  const transformInfraConfigs = () => {
+    const updatedWorkingConfigs: ConfigTransform[] = [
       {
-        provider: 'GOOGLE',
-        status: updatedConfigs?.providers.google.enabled ? 'ENABLE' : 'DISABLE',
+        config: GOOGLE_CONFIGS,
+        enabled: updatedConfigs?.providers.google.enabled,
+        fields: updatedConfigs?.providers.google.fields,
       },
       {
-        provider: 'MICROSOFT',
-        status: updatedConfigs?.providers.microsoft.enabled
-          ? 'ENABLE'
-          : 'DISABLE',
+        config: GITHUB_CONFIGS,
+        enabled: updatedConfigs?.providers.github.enabled,
+        fields: updatedConfigs?.providers.github.fields,
       },
       {
-        provider: 'GITHUB',
-        status: updatedConfigs?.providers.github.enabled ? 'ENABLE' : 'DISABLE',
+        config: MICROSOFT_CONFIGS,
+        enabled: updatedConfigs?.providers.microsoft.enabled,
+        fields: updatedConfigs?.providers.microsoft.fields,
       },
       {
-        provider: 'EMAIL',
-        status: updatedConfigs?.mailConfigs.enabled ? 'ENABLE' : 'DISABLE',
+        config: MAIL_CONFIGS,
+        enabled: updatedConfigs?.mailConfigs.enabled,
+        fields: updatedConfigs?.mailConfigs.fields,
       },
     ];
-  });
+
+    const transformedConfigs: UpdatedConfigs[] = [];
+
+    updatedWorkingConfigs.forEach(({ config, enabled, fields }) => {
+      config.forEach(({ name, key }) => {
+        if (enabled && fields) {
+          const value =
+            typeof fields === 'string' ? fields : String(fields[key]);
+          transformedConfigs.push({ name, value });
+        }
+      });
+    });
+
+    return transformedConfigs;
```

Comment:
We could rewrite this in a `reduce` based approach as below. Please feel free to push if you'd like it.

```ts
    return updatedWorkingConfigs.reduce(
      (transformedConfigs: UpdatedConfigs[], { config, enabled, fields }) => {
        if (enabled && fields) {
          config.forEach(({ name, key }) => {
            const value =
              typeof fields === 'string' ? fields : String(fields[key]);
            transformedConfigs.push({ name, value });
          });
        }
        return transformedConfigs;
      },
      []
    );
```